bc_obps/registration/tests/integration/test_operation_registration.py

```diff
@@ -0,0 +1,273 @@
+from datetime import timedelta
+
+import pytest
+from django.db.models import Q
+from model_bakery import baker
+from registration.models import RegistrationPurpose, FacilityDesignatedOperationTimeline, Operation
+from registration.tests.constants import MOCK_DATA_URL
+from registration.tests.utils.bakers import bc_obps_regulated_operation_baker
+from registration.tests.utils.helpers import CommonTestSetup, TestUtils
+from registration.utils import custom_reverse_lazy
+
+
+class TestOperationRegistration(CommonTestSetup):
+    def setup_method(self):
+        super().setup_method()
+        self.approved_user_operator = baker.make_recipe('utils.approved_user_operator', user=self.user)
+        self.bcghg_id = baker.make_recipe('utils.bcghg_id')
+        self.swrs_facility_id = 1003
+        self.boro_id = bc_obps_regulated_operation_baker()
+        # add some random contacts so we can test that they are not overridden
```

Comment:
Might be good to add a facility to test for no overwrites too--probably many users will already have facilities in the system from the SWRS import

Reply:
I think we can’t do that since we’re testing the SFO type, unless I create an additional set of tests for the LFO. 🤔

Reply:
Added another parameterized test for operation types.